ko/pl.rst

```diff
@@ -0,0 +1,1383 @@
+        <using_element> ::= [ { IN | IN OUT | OUT } ] <expression>
+        <into_clause> ::= INTO <identifier> [ , <identifier>, ... ]
+
+
+* *dynamic_sql*: 문자열 타입을 갖는 표현식. 표현식은 CUBRID SQL 규약에 맞는 SQL 구문 문자열을 계산 결과로 가져야 한다.  SQL 구문 중간중간 값을 필요로 하는 자리에 ?(물음표)를 대신 쓸 수 있으며 이러한 ?의 갯수와 *using_clause*\에 포함된 표현식의 갯수는 일치해야 한다.
+* *using_clause*: *dynamic_sql*\을 실행할 때 문자열의 ? 자리에 채워질 값들을 지정한다.  IN, IN OUT, OUT 세 가지 타입으로 지정할 수 있다.
+* *into_clause*: *dynamic_sql*\이 SELECT문을 나타내는 경우에 조회 결과를 담을 변수나 OUT 파라메터를 지정한다.
+
+다음은 EXECUTE IMMEDIATE의 사용예이다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE collect_athlete_history(name VARCHAR) AS
+        new_table VARCHAR := name || '_history';
+    BEGIN
+        EXECUTE IMMEDIATE 'drop table if exists ' || new_table;
+        EXECUTE IMMEDIATE 'create table ' || new_table || ' like history';
+        EXECUTE IMMEDIATE 'insert into ' || new_table || ' select * from history where athlete = ?'
+            USING name;
+    END;
+
+대입문
+======
+::
+
+    <assignment_statement> ::=
+        <identifier> := <expression>
+
+* *identifier*: 변수이거나 OUT 파라메터이어야 한다.
+* *expression*: 대입될 값을 계산하는 표현식. 아래 표현식 절 참조
+
+CONTINUE, EXIT
+===============
+::
+
+    <continue_statement> ::=
+        CONTINUE [ <label_name> ] [ WHEN <expression> ]
+
+::
+
+    <exit_statement> ::=
+        EXIT [ <label_name> ] [ WHEN <expression> ]
+
+
+CONTINUE와 EXIT 문은 루프문 안에서만 사용할 수 있다.
+CONTINUE 문은 아래쪽으로의 실행 흐름을 멈추고 루프의 처음으로 분기해서 다음 iteration을 실행하도록 한다.
+EXIT 문은 아래쪽으로의 실행 흐름을 멈추고 루프를 빠져나가 그 루프 다음 실행문으로 분기한다.
+*label_name*\이 없는 경우 그 CONTINUE/EXIT 문을 포함하는 가장 안쪽의 루프를 재시작한다/빠져나간다.
+루프가 여럿 중첩된 경우 *label_name*\을 지정하여 분기할 루프를 지정할 수 있다.
+WHEN 절이 있는 경우 BOOLEAN 타입의 *expression*\이 TRUE로 계산될 경우에만 분기한다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE test_proc AS
+        i INT := 0;
+    BEGIN
+        LOOP
+            put_line(i);            -- 0, 1, 2, 3, 4, 5
+            i := i + 1;
+            CONTINUE WHEN i < 3;
+            put_line(i);            -- 3, 4, 5
+            EXIT WHEN i = 5;
+        END LOOP;
+
+        put_line(i);                -- 5
+    END;
+
+NULL
+====
+::
+
+    <null_statement> ::=
+        NULL
+
+어떤 경우에는 아무 일도 하지 않는다는 것을 명시적으로 표시하고 싶을 때 NULL을 실행문으로 사용할 수 있다.
+문법상 실행문 자리를 채우기 위한 place holder 구문이다.
+
+.. _raise:
+
+RAISE
+=====
+::
+
+    <raise_statement> ::=
+        RAISE [ <identifier> ]
+
+Exception을 일으킨다.
+Exception 이름 *identifier*\가 생략되는 경우는 RAISE 문의 위치가 예외처리 구조의 THEN 절 안에 있을 때 뿐이다.
+이 경우, 현재 처리 중인 Exception을 일으키는 것으로 동작한다.
+
+.. code-block:: sql
+
+    CREATE OR REPLACE FUNCTION fibonacci(n INTEGER) RETURN INTEGER IS
+        invalid_input EXCEPTION;
+    BEGIN
+        IF n <= 0 THEN
+            RAISE invalid_input;
+        END IF;
+
+        IF n = 1 OR n = 2 THEN
+            RETURN 1;
+        ELSE
+            RETURN fibonacci(n-1) + fibonacci(n-2);
+        END IF;
+    EXCEPTION
+        WHEN invalid_input THEN
+            PUT_LINE('invalid input: ' || n);
+            RAISE;      -- 현재 처리 중인 invalid_input을 다시 일으킴
```

Comment:
Does this mean passing an invalid_input exception to the caller?

Reply:
Yes, it does.